app/services/group_message.rb

```diff
@@ -41,6 +41,21 @@ def create
     end
   end
 
+  def delete_previous!
+    posts = Post
```

Comment:
This is reasonably safe but I wonder if we could harden it more:

- Ensure the topic was created by system user
- Include a maximum date range. It's probably OK if they are warned again after a few months.